src/test/java/wooteco/subway/dao/LineDaoTest.java

```diff
@@ -0,0 +1,138 @@
+package wooteco.subway.dao;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.util.List;
+import java.util.Optional;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import wooteco.subway.domain.Line;
+import wooteco.subway.exception.line.NoSuchLineException;
+
+public class LineDaoTest extends DaoTest {
+
+    @Test
+    @DisplayName("노선을 저장하면 저장된 노선 정보를 반환한다.")
+    void Save() {
+        // given
+        final Line line = new Line("7호선", "bg-red-600");
+
+        // when
+        final Line savedLine = lineDao.insert(line).orElseThrow();
+
+        // then
+        assertThat(savedLine.getName()).isEqualTo(line.getName());
+        assertThat(savedLine.getColor()).isEqualTo(line.getColor());
+    }
+
+    @Test
+    @DisplayName("저장하려는 노선 이름이 중복이면 예외를 던진다.")
+    void Save_DuplicateName_ExceptionThrown() {
+        // given
+        final String name = "7호선";
+
+        lineDao.insert(new Line(name, "bg-red-600")).orElseThrow();
+
+        // when
+        final Line line = new Line(name, "bg-blue-600");
+
+        // then
+        final Optional<Line> possibleLine = lineDao.insert(line);
+        assertThat(possibleLine).isEmpty();
+    }
+
+    @Test
+    @DisplayName("모든 노선을 조회한다.")
+    void FindAll() {
+        // given
+        final Line line7 = new Line("7호선", "bg-red-600");
+        lineDao.insert(line7);
+
+        final String line5Name = "5호선";
+        final String line5Color = "bg-green-600";
+        final Line line5 = new Line(line5Name, line5Color);
+        lineDao.insert(line5);
+
+        // when
+        final List<Line> lines = lineDao.findAll();
+
+        // then
+        assertThat(lines.size()).isEqualTo(2);
+    }
+
+    @Test
+    @DisplayName("id를 통해 해당하는 노선을 조회한다.")
+    void FindById() {
+        // given
+        final Line line7 = new Line("7호선", "bg-red-600");
+        final Line persistLine = lineDao.insert(line7).orElseThrow();
+
+        // when
+        final Optional<Line> actual = lineDao.findById(persistLine.getId());
+
+        // then
+        assertThat(actual.isPresent()).isTrue();
+        assertThat(actual.get()).isEqualTo(persistLine);
+    }
+
+    @Test
+    @DisplayName("존재하지 않은 id로 노선을 조회하면 Empty Optional을 반환한다.")
+    void FindById_InvalidId_EmptyOptional() {
+        // when
+        final Optional<Line> possibleLine = lineDao.findById(1L);
+
+        // then
+        assertThat(possibleLine.isEmpty()).isTrue();
```

Comment:
```suggestion
assertThat(possibleLine).isEmpty()
```

이런식으로 작성이 가능하빈다 :) 한번 어떤 방식이 좋을지는 판단해주세요!